### DISCUSSION TITLE
Release 1.21

### DISCUSSION THREAD

**Discussion by webspiderteam** [webspiderteam/Bluetooth-DMM-For-Windows#27] [Announcements]
Changed MQTT Library because of cousing connection issues and crashes

<hr /><em>This discussion was created from the release <a href='https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/releases/tag/WPFRelease1.21'>Release 1.21</a>.</em>

**Comment by ludwich66** [comment]
The last MQTT test was successful with 70,000 records!

congratulations
L

**Reply by webspiderteam** [reply]
Your last test was bigger ☺️ but 20 hours is good enough. Thanks for testing and give your time to help 🙏🤝

**Comment by webspiderteam** [comment]
What do you think about that @ludwich66 

![image](https://user-images.githubusercontent.com/85828505/187747607-1b36a4d3-5b50-4a16-acf3-5c7087bb0e1f.png)

**Reply by ludwich66** [reply]
I like the menu system on first view - icon for light identification - and localized language.
Is it a moke up? or did it work?
If it usable, send me a preview 
Did you like customized icons ?

Ludwich

You are very agil ;-)

**Reply by webspiderteam** [reply]
> icon for light identification

what did you mean?

It is very helpful if the icon is already "self-explanatory", the text complements the whole thing.


> and localized language.

ofcourse I will localize Its now in constraction state. I only made this page for now.

> Did you like customized icons ?

which icons? I just find these from net. These are the best that I can found.

If you like the good everything is OK, otherwise I'm happy to extend the "BT-DMM" iconset. It's nice when the symbols all match in size and impact

> You are very agil ;-)

Not so much :)

Yes, yes
Your last words were "now" is it enough for this program ;-)

**Reply by webspiderteam** [reply]
did you just edit my comment 🤣 
I also wil put an updater. 
yes its enough but I thought  that this would be the last update and what if it needs an update and no one can know for a while later. then I think If I put a auto update feauture they will know its updated :) thats all

**Comment by ludwich66** [comment]
You still have a point unclear ;-)
The V05B can also be switched from the HandyApp.
So the stream is in two directions :-)
But I haven't done any research on that yet.

**Reply by webspiderteam** [reply]
> So the stream is in two directions :-)

yes we can send data but which data which format and we dont know what to send. there is also another gatt id that is only for sending data to multimeter but we cant know until we see some data

**Reply by webspiderteam** [reply]
And also there is a problem. My dmm is not accepting any data and closes even with original app.

**Reply by ludwich66** [reply]
I know your 9002. I also had one that I modified for Bluetooth decryption.
I then shot it when determining the low-batt values on the laboratory power supply. Unfortunately I couldn't find a repairable error. The bloob no longer works.
I replaced it with V005B.

**Reply by webspiderteam** [reply]
I did same mistake with my old multimeter :)

**Comment by webspiderteam** [comment]
![image](https://user-images.githubusercontent.com/85828505/188173060-65721b7e-ae0e-4ab8-9936-12ca92de3730.png)
updates page

**Comment by ludwich66** [comment]
Oh nice - it shows direct packages from Github?

L

**Reply by webspiderteam** [reply]
yes from github and can directly download without opening browser. Not self updater but thats enough I think.

**Reply by webspiderteam** [reply]
of course I will add save button later :D

**Reply by webspiderteam** [reply]
Please give me an idea. where should I put notification of "new update available"?

**Reply by ludwich66** [reply]
Is it possible in Main screen 
"DMM-Name: connected" 2sec. "Update available" 
Change in 2 Second rythm?

**Reply by webspiderteam** [reply]
My main worries are if its 2 or 5 seconds user can miss it. if it will longer it can annoy user. I can put it in menu but this time maybe user wont open menu. if I put a popup it will push user to respond. 
All my thinks are have cons.

**Reply by ludwich66** [reply]
I meant that the main screen (is normally the only thing that would be visible).
It would even be enough if it happened at startup via "Update Info Box". It is certainly not necessary to update while a measurement is running.
L

**Reply by webspiderteam** [reply]
my posible solution is putting it under main display and below chart... and showing it until connection and 5 second more. what do you think?

**Comment by ludwich66** [comment]
Ja Versuchen Sie es ;-)

L

**Reply by webspiderteam** [reply]
Its kind of ready I think. Do you wanna test drive? :)
[Portable.zip](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/9482485/Portable.zip)

**Reply by ludwich66** [reply]
I do It!! - after my CoffeeTime ;-)
CU!
L

**Comment by ludwich66** [comment]
Shortys: from top to Bottom
Language Box need more space
Set MQTT Values in another Frame in second place or move it into box one
( clear separation from standard mqtt credentials )

L: Only for my Thinking: Auto Reconnect after Disconnection vs. Auto Connect Known Devices on Startup

**Updates:** Looks good, but i mean it ist not logical if i see on first Page the Update-Banner (+++) and i must manual search with "Check Updates"
--> is Checkbox "Check Updates on Startup" set, then display result instantly (Set As Preset)
--> is Checkbox "Check Updates on Startup" NOT set then do it as now

I like this Boxinterface ;-)

L

I think this great change deserves a versionnumber 2!

**Reply by webspiderteam** [reply]
> i must manual search with "Check Updates"

you dont have to. its already checking automaticaly. If you wisk to check manually you will see all info



> is Checkbox "Check Updates on Startup" set, then display result instantly (Set As Preset)

Its default enabled. Its not updating itself thats why I did it like that. Just warns you. If I put a self update it needs loader app.

**Reply by ludwich66** [reply]
hello Engin, we're talking past each other ;-)
I wanted to say that if I get the red notification, then the version information should also be visible under "Updates" - it is not necessary to manually press the button again.
It is only necessary for users who do not want auto-updates:
They have not the Red Info before! They made all manually.
L

**Reply by webspiderteam** [reply]
> I wanted to say that if I get the red notification, then the version information should also be visible under "Updates" - it is not necessary to manually press the button again.

you dont need it. probably you couldnt see it but its already doing what you said.  it was just latest version. 
[Debug.zip](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/9482965/Debug.zip) take this you will see ı changed version of app from 1.21 to 120

**Reply by webspiderteam** [reply]
- [x] Language Box
- [x] Show Update info if auto checked (Has it already)
- [ ] Values move (has problem with it) I will leave it for now  

> L: Only for my Thinking: Auto Reconnect after Disconnection vs. Auto Connect Known Devices on Startup

there is simple difference. If you have more than 1 device and you connected one of them and connection lost 
1. if Reconnect checked and Ask Before connect App will not ask for connect same device directly connect same device
2. if reconnect not checked and ask before connect checked app will ask to connect every time even it is same device.
3. if Auto connect startup not checked you have to manually connect from Connect to device
a. if Reconnect not checked you have to reconnect device manually.
b. if reconnect checked it will reconnect same device but no other device.
